### ISSUE TITLE
Binance Perpetual - Error fetching funding income

### ISSUE THREAD

**Issue by RC-13** [CoinAlpha/hummingbot#3226]
**Describe the bug**
Opening ticket for tracking: The issue is quite difficult to replicate since we have to wait for the funding payment. During this period, the bot encounters the following error:
```ruby
2021-04-19 08:00:00,941 - 2327 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Error fetching /fapi/v1/income
Traceback (most recent call last):
  File "/home/ubuntu/binance_perpetual/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 1016, in request
    error_response = await response.json()
  File "/home/ubuntu/miniconda3/envs/hummingbot/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 1026, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://fapi.binance.com/fapi/v1/income?incomeType=FUNDING_FEE&limit=1&recvWindow=20000&symbol=ETHUSDT&timestamp=1618819200000&signature=588efa81ddfe71cfdf89c099fc050a9b60936a5028ed77a5096e7d09f1947bbd')
2021-04-19 08:00:00,941 - 2327 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - WARNING - 0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://fapi.binance.com/fapi/v1/income?incomeType=FUNDING_FEE&limit=1&recvWindow=20000&symbol=ETHUSDT&timestamp=1618819200000&signature=588efa81ddfe71cfdf89c099fc050a9b60936a5028ed77a5096e7d09f1947bbd')
2021-04-19 08:00:00,942 - 2327 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Unexpected error in user stream listener loop: 'ContentTypeError' object is not subscriptable
Traceback (most recent call last):
  File "/home/ubuntu/binance_perpetual/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 559, in _user_stream_event_listener
    await self.get_funding_payment()
  File "/home/ubuntu/binance_perpetual/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 942, in get_funding_payment
    payment = Decimal(funding_payment["income"])
TypeError: 'ContentTypeError' object is not subscriptable
```
The current open position and exit order remain unaffected.

**Steps To Reproduce**
Here are the steps to reproduce the issue (see attachments in the section below):
1.  Create a perpetual market making strategy using Binance Perpetual
2. Set either one-way or hedge mode and enable profit-taking position management
3. Leave your bot running until you have an open position then wait until the funding period occurs.
...

**Screenshots**
![binance-perp](https://user-images.githubusercontent.com/57189990/115221664-7cde8680-a13c-11eb-8604-9998f8a33eb9.png)

**Release version**
0.38.0 running via Source build

**Attachments**
[logs_perpetual-ETH_USDT.log](https://github.com/CoinAlpha/hummingbot/files/6335193/logs_perpetual-ETH_USDT.log)

[logs_perpetual-ETH_USDT-2021-04-17.log](https://github.com/CoinAlpha/hummingbot/files/6335200/logs_perpetual-ETH_USDT-2021-04-17.log)

*WARNING: Do NOT publish any exchange API keys or your wallet's private key. Whoever has access to them may steal your assets!*

**Optional: your discord username:**

**Comment by RC-13** [comment]
After renaming the SQLite file, the error hasn't appeared for 2 days now. Will continue to monitor the bot and close the ticket if the issue will no longer persist.

**Comment by rxlxrxsx** [comment]
Confirming the bug appears as well when using spot_perpetual_arbitrage.

```
2021-05-12 00:00:00,256 - 1 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Error fetching /fapi/v1/income
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 1016, in request
    error_response = await response.json()
  File "/home/hummingbot/miniconda3/envs/hummingbot/lib/python3.8/site-packages/aiohttp/client_reqrep.py", line 1026, in json
    raise ContentTypeError(
aiohttp.client_exceptions.ContentTypeError: 0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://testnet.binancefuture.com/fapi/v1/income?incomeType=FUNDING_FEE&limit=1&recvWindow=20000&symbol=BTCUSDT&timestamp=1620777600000&signature=720e891b5bb5d6163c5b5cf39d4f2840ce92742fbae6669ec474d769024f3bbd')
2021-05-12 00:00:00,259 - 1 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - WARNING - 0, message='Attempt to decode JSON with unexpected mimetype: ', url=URL('https://testnet.binancefuture.com/fapi/v1/income?incomeType=FUNDING_FEE&limit=1&recvWindow=20000&symbol=BTCUSDT&timestamp=1620777600000&signature=720e891b5bb5d6163c5b5cf39d4f2840ce92742fbae6669ec474d769024f3bbd')
2021-05-12 00:00:00,260 - 1 - hummingbot.connector.derivative.binance_perpetual.binance_perpetual_derivative - ERROR - Unexpected error in user stream listener loop: 'ContentTypeError' object is not subscriptable
Traceback (most recent call last):
  File "/home/hummingbot/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 559, in _user_stream_event_listener
    await self.get_funding_payment()
  File "/home/hummingbot/hummingbot/connector/derivative/binance_perpetual/binance_perpetual_derivative.py", line 942, in get_funding_payment
    payment = Decimal(funding_payment["income"])
TypeError: 'ContentTypeError' object is not subscriptable
```

Version: development-0.39.0
Logs/config: [spot_perp_bug.zip](https://github.com/CoinAlpha/hummingbot/files/6462630/spot_perp_bug.zip)

**Comment by dennisocana** [comment]
Fixed in version 0.41 (July) release by @zappra - thanks for contributing!
